src/man/sssd-idp.5.xml

```diff
@@ -0,0 +1,233 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE reference PUBLIC "-//OASIS//DTD DocBook V4.5//EN"
+"http://www.oasis-open.org/docbook/xml/4.5/docbookx.dtd">
+<reference>
+<title>SSSD Manual pages</title>
+<refentry>
+    <xi:include xmlns:xi="http://www.w3.org/2001/XInclude" href="include/upstream.xml" />
+
+    <refmeta>
+        <refentrytitle>sssd-idp</refentrytitle>
+        <manvolnum>5</manvolnum>
+        <refmiscinfo class="manual">File Formats and Conventions</refmiscinfo>
+    </refmeta>
+
+    <refnamediv id='name'>
+        <refname>sssd-idp</refname>
+        <refpurpose>SSSD IdP provider</refpurpose>
+    </refnamediv>
+
+    <refsect1 id='description'>
+        <title>DESCRIPTION</title>
+        <para>
+            This manual page describes the configuration of the IdP provider
+            for
+            <citerefentry>
+                <refentrytitle>sssd</refentrytitle>
+                <manvolnum>8</manvolnum>
+            </citerefentry>.
+            For a detailed syntax reference, refer to the <quote>FILE FORMAT</quote> section of the
+            <citerefentry>
+                <refentrytitle>sssd.conf</refentrytitle>
+                <manvolnum>5</manvolnum>
+            </citerefentry> manual page.
+        </para>
+        <para>
+            The IdP provider is a back end used to connect to an OAuth 2.0 and
+            REST based identity provider (IdP). Since products might have
+            individual implementation of the REST API for looking up user and
+            group attributes dedicated code might be required, see the
+            <quote>idp_type</quote> option for details.
+        </para>
+        <para>
+            IdPs typically do not provide POSIX attributes like e.g.
+            user Id (UID) or home directory. SSSD's IdP provider will
+            autogenerate the needed attributes. The default algorithm to
+            generate user IDs (UIDs) and group IDs (GIDs) aims to create
+            reproducible IDs on different systems. As a drawback it might
+            happen that the algorithm assigns the same ID to different objects
+            and only the first one requested via SSSD will be available.
```

Comment:
I didn't get this part - "only the first one requested via SSSD will be available".

Reply:
The second object (user or group) which gets the same ID and causes the collision cannot be saved into the cache, so SSSD only knows the first one. Do you have a suggestion for a better sentence?

Reply:
Yes, it depends on the lookup order. The id-mapping library allows to add other algorithms and when we have more experience/feedback how the IdPs are used we can add others which can avoid collisions, ideally just reading the POSIX ID from the IdP.

With the current algorithm lookups by-id only work for objects already in the cache.

Reply:
> > when we have more experience/feedback how the IdPs are used we can add others which can avoid collisions
> 
> This means id-mapping can change after SSSD update. Worth mentioning in the release notes (upstream and Fedora).

No, the idea is to keep the current one as default, since it is something that should work in all scenarios, and allow to use other schemes which the help of new options to change the behavior of the POSIX id-mapping library.